GUI/SearchTaskWindow.xaml.cs

```diff
@@ -144,6 +144,7 @@ private void PopulateChoices()
 
         private void UpdateFieldsFromTask(SearchTask task)
         {
+            proteaseComboBox.SelectedItem = task.CommonParameters.DigestionParams.SpecificProtease; //needs to be first, or nonspecific overwrites for missed cleavages/max length
```

Comment:
I don't understand why this is moved

Reply:
For non-specific searches with nonspecific protease, the number of missed cleavages is always updated as the maximum length -1 and vice versa. If the protease is not actually non-specific (like you're doing chymotrypsin), but you want to look for non-specific peptides (because chymotrypsin is awful), then you can do a non-specific search, but keep chymotrypsin as a the protease for FDR purposes. If protease isn't set first, then the missed cleavages is updated to the maximum length -1 (50-1=49). That would make sense in nonspecific searches, but we interpret it as allowing 49 chymotrypsin cleavages, which is an insane number.

Would you like me to make that explanation a comment?